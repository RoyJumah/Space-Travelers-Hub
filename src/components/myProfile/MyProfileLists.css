li {
  border-bottom: 1px solid black;
  padding: 10px;
  list-style: none;
}

h2 {
  margin-bottom: 12px;
}

.myprofile__list {
  border: 1px solid black;
}
