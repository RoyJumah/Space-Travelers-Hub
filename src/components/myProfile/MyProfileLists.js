/* eslint-disable react/prop-types */
import { useSelector } from 'react-redux';
import { selectAllRockets } from '../../redux/rockets/rocketsSlice';
import { selectAllMissions } from '../../redux/missions/missionsSlice';

const RocketItem = ({ rocketName }) => <li>{rocketName}</li>;

const MissionItem = ({ missionName }) => <li>{missionName}</li>;

const MyProfileLists = () => {
  const rockets = useSelector(selectAllRockets);
  const missions = useSelector(selectAllMissions);
  const reservedRockets = rockets.filter((rocket) => rocket.reserved === true);
  const joinedMissions = missions.filter((mission) => mission.joined === true);
  return (
    <>
      <section>
        <h2>My Missions</h2>
        <ul>
          {joinedMissions.map((mission) => (
            <MissionItem
              key={mission.mission_id}
              missionName={mission.mission_name}
            />
          ))}
        </ul>
      </section>
      <section>
        <h2>My Rockets</h2>
        <ul>
          {reservedRockets.map((rocket) => (
            <RocketItem key={rocket.id} rocketName={rocket.name} />
          ))}
        </ul>
      </section>
    </>
  );
};

export default MyProfileLists;
