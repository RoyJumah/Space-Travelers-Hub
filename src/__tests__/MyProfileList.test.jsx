import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import MyProfileLists from '../components/myProfile/MyProfileLists';
import { Provider } from 'react-redux';
import store from '../redux/ConfigureStore';


describe('Display myProfileList component', () => {
  test('it matches snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MyProfileLists />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })

  test('should return  MyProfileList Component', () => {
    render(
      <Provider store={store}>
        <MyProfileLists />
      </Provider>
    );
  })
})
