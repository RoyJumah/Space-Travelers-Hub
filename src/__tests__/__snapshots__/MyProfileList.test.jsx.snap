// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display myProfileList component it matches snapshot 1`] = `
Array [
  <section>
    <h2>
      My Missions
    </h2>
    <ul />
  </section>,
  <section>
    <h2>
      My Rockets
    </h2>
    <ul />
  </section>,
]
`;
